### PR TITLE
Add disposal mechanism for created Layers to TestRecordingPaintingContext.

### PR DESCRIPTION
**Pull request by polina-c**
Contributes to https://github.com/flutter/flutter/issues/134575

### CODE REVIEW

**Comment by polina-c** [comment]
The PR is not test covered as it is first step to fix https://github.com/flutter/flutter/issues/134575, that is test covered.

**Comment by Hixie** [comment]
test-exempt: preparing for extra testing

**Review by hannah-hyj — APPROVED** [review]
LGTM

File: packages/flutter_test/lib/src/recording_canvas.dart [hunk 1]
```diff
@@ -170,7 +173,18 @@ class TestRecordingPaintingContext extends ClipContext implements PaintingContex
     canvas.saveLayer(null, Paint()); // TODO(ianh): Expose the alpha somewhere.
     painter(this, offset);
     canvas.restore();
-    return OpacityLayer();
+    final OpacityLayer layer = OpacityLayer();
+    _createdLayers.add(layer);
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/recording_canvas.dart:177]: As I said on https://github.com/flutter/flutter/pull/134701#discussion_r1326534750, this is likely not the right fix. If I remember correctly, the contract of PaintingContext (which defines this method) is that the caller assumes ownership of the Layer and is in charge of disposing it. If this change is in fact "fixing" any leaks, it is only doing so for the test. In production (where a regular PaintContext and not the testing version is used) this leak is still present!

**Pull request merged**